packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -46,9 +84,33 @@ export interface DecorateQueryProcedure<TDef extends ResolverDef> {
    * @see https://tanstack.com/query/latest/docs/framework/react/reference/infiniteQueryOptions#infinitequeryoptions
    */
   infiniteQueryOptions: TRPCInfiniteQueryOptions<TDef>;
+
+  /**
+   * Calculate the Tanstack Query Key for a Query Procedure
+   *
+   * @see https://tanstack.com/query/latest/docs/framework/react/guides/query-keys
+   */
+  queryKey: (input?: TDef['input']) => TRPCQueryKey;
+
+  /**
+   * Calculate a Tanstack Query Filter for a Query Procedure
+   *
+   * @see https://tanstack.com/query/latest/docs/framework/react/guides/filters
+   */
+  queryFilter: (input?: TDef['input']) => QueryFilters;
 }
 
 export interface DecorateMutationProcedure<TDef extends ResolverDef> {
```

Comment:
should this have `.mutationKey()` ?

Reply:
I think that's a different user story to queryKey really, I'm not very familiar with mutationKey so wouldn't want to rush into adding that without knowing the story, but it seems to be more about remotely executing a mutation which wouldn't make a lot of sense with tRPC

Reply:
It's needed for `queryClient.isMutating()` for example, a utility we currently have

Reply:
Got it, are you okay if we implement that as additional work into your branch? The scope of this PR already expanded a lot and it was all purely in service of polymorphism support. mutationKey isn't needed for that case

Reply:
Wait, what's the correct branch? I thought I had it from your PR?

Reply:
This is Alex's R19 branch which I think isn't working in some areas 🤔

Reply:
Crap. Well that threw me for a loop and into git hell: https://github.com/trpc/trpc/pull/6244

Reply:
Closing this in favour of the other MR